robottelo/ui/computeresource.py

```diff
@@ -441,3 +442,23 @@ def set_profile_values(self, resource_name, profile_name, **kwargs):
             self, resource_type)
         resource_profile_form.set_values(**kwargs)
         resource_profile_form.submit()
+
+    def check_image_os(self, os_name):
+        """Check if the OS is present, if not create the required OS
+
+        :param os_name: OS name to check, and create
+        :return Created os
+        """
+        # Check if OS that image needs is present or no, If not create the OS
+        result = entities.OperatingSystem().search(query={
+            u'search': u'title="{0}"'.format(os_name)
+        })
+        os = entities.OperatingSystem().create()
+        if len(result) == 0 or (os_name != result[0].title):
+            os = entities.OperatingSystem(
+                id=os.id,
+                name=os_name.split(' ')[0],
+                major=os_name.split(' ')[1].split('.')[0],
+                minor=os_name.split(' ')[1].split('.')[1],
+            ).update(['name', 'major', 'minor'])
+        return os
```

Comment:
Yes, we can do that,  but we have to be sure of deleting the Default OS(Red Hat 7.3 ) , if we create it. And if we use the approach you have mentioned and if the Default OS is already being used by some hosts or associated with resources , we won't be able to delete this.

> P.S  OS is global and not limited to Org's

Now , Why we need to delete the `created` Default OS (Red Hat 7.3) ?
This is because, the default OS is auto-populated by satellite and has all the associations, like provisioning templates , kickstarts , etc.
The OS that we create will not have those, and if satellite tries to create the default os it will fail as we would have created the os with same name.
The problem, we are facing is, 
1) Some other test is deleting the Default OS  (Red Hat 7.3).
2) There are 4 image based tests of which 3 are run , before the satellite auto-populates the Default OS  (Red Hat 7.3).
3) The 4th test is a success, because by the time the 4th test is run, Satellite auto-populates the required Default OS  (Red Hat 7.3).

So, the approach,
1) Create a random OS
2) Check whether we have a default OS. If yes , do nothing
3) If no, update the random os to default os
4) Delete the os  (In both if and else we are able to get and delete a OS)